src/book/book.rs

```diff
@@ -231,6 +267,31 @@ fn load_chapter<P: AsRef<Path>>(link: &Link, src_dir: P) -> Result<Chapter> {
     Ok(ch)
 }
 
+fn load_virtual_chapter<P: AsRef<Path>>(link: &VirtualLink, src_dir: P) -> Result<VirtualChapter> {
+    let src_dir = src_dir.as_ref();
+
+    let content = String::new();
+        /* TODO Open question: Do we need "pseudo content" for virtual chapters?
+         * If not: Remove this and corresponding field in `VirtualChapter`!
+         */
+
+    let mut ch = VirtualChapter::new(&link.name, content);
+    ch.number = link.number.clone();
+
+    let sub_items = link.nested_items
+        .iter()
+        .map(|i| load_summary_item(i, src_dir))
+        .collect::<Result<Vec<_>>>()?;
+
+    ch.sub_items = sub_items;
+
+    Ok(ch)
+        /* TODO Open question: Do we really want to return a `Result<_>` here?
+         * Pro: Function would have almost signature as load_chapter()
+         * Con: We do not use it!
```

Comment:
But we do use it. When you call `load_summary_item` recursively on the sub-items you return an error if any of the sub-items error.

Reply:
You're right. I overlooked the `?`-operator in line 284. Thanks!